src/app/services/tvdb.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { TvShowData } from '../tvshows/store/tvshows.models';
+
+const TVSHOWS: TvShowData[] = [
+  {title: 'WalkingDead'},
+  {title: 'Wheel of Forune'},
+  {title: 'The Walking Dead'},
+  {title: 'Rick & Morty'},
+];
+
+@Injectable()
+export class TvDbService {
+  search(query: string): Promise<any> {
```

Comment:
Curious why you choose to use a promise here. Also why `<any>` and not a typed shape `Promise<TvDataData[]>`?

Reply:
I thought a Promise is the appropriate way to handle API fetch requests, if the connection isn't left open.  

It should be Promise<TvShowData>, will fix.

Reply:
Since it's an Angular app you should try and use Observable's as much as possible. Therefore, I'd rather see this return `Observable<vShowData[]>` instead of a promise.